src/harness/util.py

```diff
@@ -45,76 +52,67 @@ def _getRandomLatLongInPolygon(polygon):
-                                      previous_year_date.year)
-                           if index == 1 else str(val) for index, val in
-                                    enumerate(pal_record['palId'].split('/'))])
-    pal_record['userId'] = device['userId']
+    pal_rec['palId'] = '/'.join(['pal', '%s-%d' % ('{:02d}'.format(previous_year_date.month),
+                                                   previous_year_date.year), str(ppa_fips_code),
+                                 fcc_channel_id])
+    pal_rec['userId'] = user_id
     # Make the date consistent in Pal Record for Registration and License
-    pal_record = _changeDateInPpaAndPal(pal_record,
-                                        previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                                        next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+    pal_rec['registrationInformation']['registrationDate'] = \
+      previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+
     # Change License Information in Pal
-    pal_record['license']['licenseAreaIdentifier'] = str(ppa_fips_code)
-    pal_record['license']['licenseAreaExtent'] = \
+    pal_rec['license']['licenseAreaIdentifier'] = str(ppa_fips_code)
+    pal_rec['license']['licenseAreaExtent'] = \
       'zone/census_tract/census/%d/%d' % (ppa_census_year, ppa_fips_code)
+    pal_rec['license']['licenseDate'] = previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+    pal_rec['license']['licenseExpiration'] = next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+    pal_rec['license']['licenseFrequencyChannelId'] = fcc_channel_id
     # Change Frequency Information in Pal
-    for assign in ('primaryAssignment', 'secondaryAssignment'):
-      pal_record['channelAssignment'][assign]['lowFrequency'] = low_frequency
-      pal_record['channelAssignment'][assign]['highFrequency'] = high_frequency
-  # Make the date consistent in Ppa Record
-  ppa_record = _changeDateInPpaAndPal(ppa_record,
-                                      previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                                      next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+    pal_rec['channelAssignment']['primaryAssignment']['lowFrequency'] = low_frequency
+    pal_rec['channelAssignment']['primaryAssignment']['highFrequency'] = high_frequency
+    pal_records[index] = json.loads(json.dumps(pal_rec))
   # Add Pal Ids into the Ppa Record
-  ppa_record['ppaInfo']['palId'] = [pal_record['palId'] for pal_record
-                                    in pal_records]
-  device['installationParam']['latitude'], device['installationParam']['longitude'] = \
-    _getRandomLatLongInPolygon(ppa_record['zone'])
+  ppa_record = defaultdict(lambda: defaultdict(dict), ppa_record)
 
-  return ppa_record, pal_records, device
+  ppa_record['ppaInfo']['palId'] = [pal['palId'] for pal in pal_records]
+  ppa_record['id'] = 'zone/ppa/admin_0/%s' % (ppa_record['ppaInfo']['palId'][0])
```

Comment:
You probably want to add a random number (or other identifier here) -- otherwise injecting two PPAs which have the same PAL license (which is totally fine based on the specs) would cause the first to be overwritten by the second.